src/main/java/com/materialsoftherift/motr/datagen/MotrModelProvider.java

```diff
@@ -353,21 +352,77 @@ private void registerDirectionalSlabModel(
-                        .select(SlabType.DOUBLE, Variant.variant().with(VariantProperties.MODEL, cube))));
+        blockModels.blockStateOutput.accept(
+                MultiVariantGenerator.multiVariant(slab)
+                        .with(PropertyDispatch.property(BlockStateProperties.SLAB_TYPE)
+                                .select(SlabType.BOTTOM, Variant.variant().with(VariantProperties.MODEL, bottom))
+                                .select(SlabType.TOP, Variant.variant().with(VariantProperties.MODEL, top))
+                                .select(SlabType.DOUBLE, Variant.variant().with(VariantProperties.MODEL, cube)))
+        );
+    }
+
+    private void registerDirectionalPathSlabModel(
+            BlockModelGenerators blockModels,
+            Block slab,
+            String sideTex,
+            String topTex,
+            String bottomTex) {
+
+        TextureMapping mapping = new TextureMapping()
+                .put(TextureSlot.SIDE, ResourceLocation.withDefaultNamespace("block/" + sideTex))
+                .put(TextureSlot.TOP, ResourceLocation.withDefaultNamespace("block/" + topTex))
+                .put(TextureSlot.BOTTOM, ResourceLocation.withDefaultNamespace("block/" + bottomTex));
+
+        ResourceLocation bottom = ExtendedModelTemplateBuilder.builder()
+                .parent(MaterialsOfTheRift.id("block/path_slab_bottom"))
+                .requiredTextureSlot(TextureSlot.SIDE)
+                .requiredTextureSlot(TextureSlot.TOP)
+                .requiredTextureSlot(TextureSlot.BOTTOM)
+                .build()
+                .create(slab, mapping, blockModels.modelOutput);
+
+        ResourceLocation top = ExtendedModelTemplateBuilder.builder()
+                .parent(MaterialsOfTheRift.id("block/path_slab_top"))
+                .requiredTextureSlot(TextureSlot.SIDE)
+                .requiredTextureSlot(TextureSlot.TOP)
+                .requiredTextureSlot(TextureSlot.BOTTOM)
+                .build()
+                .createWithSuffix(slab, "_top", mapping, blockModels.modelOutput);
+
+        ResourceLocation cube = ExtendedModelTemplateBuilder.builder()
+                .parent(MaterialsOfTheRift.id("block/path_slab_double"))
+                .requiredTextureSlot(TextureSlot.SIDE)
+                .requiredTextureSlot(TextureSlot.TOP)
+                .requiredTextureSlot(TextureSlot.BOTTOM)
+                .build()
+                .createWithSuffix(slab, "_double", mapping, blockModels.modelOutput);
+
+        blockModels.blockStateOutput.accept(
+                MultiVariantGenerator.multiVariant(slab)
+                        .with(PropertyDispatch.property(BlockStateProperties.SLAB_TYPE)
+                                .select(SlabType.BOTTOM, Variant.variant().with(VariantProperties.MODEL, bottom))
+                                .select(SlabType.TOP, Variant.variant().with(VariantProperties.MODEL, top))
```

Comment:
Just noting a difference between dirt_path and dirt_path_slab blocks. Looking at it dirt_path has four variants which rotate the top texture, which appear to be are applied randomly (not based on block placement). This isn't occurring for dirt path slabs.